### ISSUE TITLE
Prometheus metrics improvements

### ISSUE THREAD

**Issue by fjetter** [dask/distributed#7345]
This is a meta ticket to collect feedback and tickets corresponding to prometheus metrics

- [x] https://github.com/dask/distributed/issues/7307
- [ ] https://github.com/dask/distributed/issues/7308
- [ ] https://github.com/dask/distributed/issues/7217
- [x] https://github.com/dask/distributed/issues/7351
- [ ] Expose https://github.com/dask/distributed/pull/7332
- [x] https://github.com/dask/distributed/issues/7341
- [x] https://github.com/dask/distributed/issues/7123
- [ ] https://github.com/dask/distributed/issues/7357
- [ ] https://github.com/dask/distributed/issues/7107

**Comment by crusaderky** [comment]
We currently measure cluster-wide utilisation over time using the spot measure of number of executing tasks on the workers. This means that with a sample time of e.g. 5s you may not notice imperfect cluster occupation if your tasks are shorter than 5s.
It should be changed to a monotonically-increasing number of tasks that are currently executing + tasks that exited computation (finished, erred, or rescheduled).

**Comment by fjetter** [comment]
cc @jacobtomlinson in case you have some thoughts / always wanted to see something specific exposed.

**Comment by jacobtomlinson** [comment]
Thanks for the ping. Nothing comes to mind at the moment but as we spin up more benchmarking work we may think of things.

One side topic that comes to mind is that we are feeling pressure within NVIDIA to instrument code with the [OpenTelemetry SDK](https://opentelemetry.io/docs/instrumentation/python/) rather than using the prometheus SDK. The industry seems to be standardising around OTel which supports metrics, tracing, logging and more. From my understanding, OTel can export prometheus metrics but also has support for pushing to other systems like Jager and Zipkin.

I need to continue doing more research but it looks pretty cool. We might want to consider making the same SDK transition in Dask so that we have more monitoring options in the future. I think the lift to switch what we have like-for-like to using the OTel SDK would be a small lift.

**Comment by fjetter** [comment]
Yes, I am aware of OTel and am open to switching. I would hope that a migration would not be too expensive.
IIUC we could even just continue writing prometheus exporters and hook it up to OTel https://opentelemetry.io/docs/instrumentation/python/exporters/#prometheus

I suggest opening another issue for this to discuss. Some of the work we need to do to support the above mentioned metrics is to collect them internally first anyhow. How we export it is almost an afterthought.
